### PR TITLE
Multiple schools

### PR DESCRIPTION
**Pull request by Daanra**
This is a very large MR, sorry about that.

This MR adds the following things:

- Authentication system. Users now have roles, and we can grant users permissions to perform certain actions or deny them. 
- Properly scales the images of lecturers. The dimensions will no longer be skewed.
- Adds a controller for forbidden pages.
- Adds cookie based sessions. Now when the server crashes or reboots, you will still be logged in :)
- Fixes PUT requests for announcements and generalinfo. These were no longer working because Chrome decided to block all PUT requests that contained HTML encoding and \n for security reasons.
- Adds a category to general information.
- Adds more functionality to the option page to add user rights and schools and such.
- Login codes are now associated with schools.
- A user with the role coordinator is associated with a school.
- Changes the GET API of a login code to return the school associated with it. 
- Fixes several small bugs and front-end issues.
- Fixes indentation at all files and adds JSDoc here and there.
- Adds a created date to general information.

### CODE REVIEW

**Comment by Daanra** [comment]
This is a test

**Comment by Daanra** [comment]
This is a test

File: config/database.js [hunk 1]
```diff
@@ -1,18 +1,62 @@
...
+	}
+	if (req.isAuthenticated())
+		return next();
+
+	res.redirect('/');
+};
 
-	    res.redirect('/');
+/**
+ * A helper function to check if all rights in an array are satisfied.
```
Comment (NikolasHadjipanayi) [inline config/database.js]: Perhaps this needs to be in another file? The rights are not really concerned with the database.

Comment (Daanra) [reply config/database.js]: I agree that the added code isn't directly related to the database. However, nothing in the file is related to the database. It's all authorization and authentication. I believe it should be in this file but the file really needs to be renamed. Could you do that?

Comment (NikolasHadjipanayi) [reply config/database.js]: Yes ill rename it to authorization

**Pull request merged**